libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -837,7 +837,7 @@ void AP_GPS::update_primary(void)
 {
 #if defined(GPS_BLENDED_INSTANCE)
     // if blending is requested, attempt to calculate weighting for each GPS
-    if (_auto_switch == 2) {
+    if ((GPS_AUTO_SWITCH)_auto_switch.get() == GPS_AUTO_SWITCH::BLEND) {
```

Comment:
Definitely not your fight - but this looks like a switch statement in hiding.

Reply:
agreed, that lingering non-switch-worthy if-statement after this is the only reason why I didn't already do it. I didn't want to change the behavior (yet)